src/container.c

```diff
@@ -224,6 +224,55 @@ static int container_setup_mount(struct hyper_container *container)
 	return 0;
 }
 
+static int container_recreate_file(char *filename)
+{
+	fprintf(stdout, "recreate file %s\n", filename);
+	hyper_unlink(filename);
+	return hyper_create(filename);
+}
+
+static int container_recreate_symlink(char *oldpath, char *newpath)
+{
+	fprintf(stdout, "recreate symlink %s to %s\n", newpath, oldpath);
+	hyper_unlink(newpath);
+	return hyper_symlink(oldpath, newpath);
+}
+
+/*
+ * Docker uses the init layer to protect against unwanted side effects on
+ * the rw layer. We recreate the same files here to have similar effect.
+ * Docker also creates directories like /dev/pts, /dev/shm, /proc, /sys,
+ * which we have over ridden in container_setup_mount, so no need to create
+ * them here.
+ */
+static int container_setup_init_layer(struct hyper_container *container)
+{
+	if (!container->initialize)
+		return 0;
+
+	if (container_recreate_file("/.dockerinit") < 0)
+		return -1;
+
+	if (container_recreate_file("/.dockerenv") < 0)
+		return -1;
+
```

Comment:
`"/.dockerinit"` and `"/.dockerenv"` were introduced for lxc driver in docker,
`"/.dockerinit"` was already removed, `"/.dockerenv"` might be removed in future,
or might be used for other purpose.

we don't need these files.